File: src/IsbInterop/IRequisiteContainer.vb
Imports IsbInterop.Data

''' <summary>
''' Интерфейс контейнера реквизитов.
''' </summary>
Friend Interface IRequisiteContainer
    ''' <summary>
    ''' Получить реквизит.
    ''' </summary>
    ''' <param name="requisiteName">Имя реквизита.</param>
    ''' <param name="getRequisiteCallback">Callback для получения реквизита.</param>
    ''' <returns>Реквизит.</returns>
    Function GetRequisite(requisiteName As String, getRequisiteCallback As Func(Of String, IRequisite)) As IRequisite

    ''' <summary>
    ''' Освободить реквизиты.
    ''' </summary>
    Sub DisposeRequisites()
End Interface

File: src/IsbInterop.CSharp/IRequisiteContainer.cs
using IsbInterop.Data;
using System;

namespace IsbInterop
{
  /// <summary>
  /// Интерфейс контейнера реквизитов.
  /// </summary>
  internal interface IRequisiteContainer
  {
    /// <summary>
    /// Получить реквизит.
    /// </summary>
    /// <param name="requisiteName">Имя реквизита.</param>
    /// <param name="getRequisiteCallback">Callback для получения реквизита.</param>
    /// <returns>Реквизит.</returns>
    IRequisite GetRequisite(string requisiteName, Func<string, IRequisite> getRequisiteCallback);

    /// <summary>
    /// Освободить реквизиты.
    /// </summary>
    void DisposeRequisites();
  }
}
